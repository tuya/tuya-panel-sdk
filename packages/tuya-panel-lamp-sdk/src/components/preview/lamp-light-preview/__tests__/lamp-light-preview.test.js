import React from 'react';
import { shallow } from 'enzyme';
import LampLightPreview from '../index';

describe('LampLightPreview components', () => {
  it('basic render', () => {
    const wrapper = shallow(<LampLightPreview workMode="white" bright={800} temperature={0} />);
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({
      workMode: 'colour',
      colour: {
        hue: 180,
        saturation: 800,
        value: 1000,
      },
    });
    wrapper.setProps({
      workMode: 'scene',
      sceneColors: [
        {
          hue: 180,
          saturation: 800,
          value: 1000,
        },
      ],
    });
    wrapper.setProps({
      workMode: 'scene',
      sceneColors: [
        {
          bright: 1000,
          temperature: 200,
        },
      ],
    });
  });
});
