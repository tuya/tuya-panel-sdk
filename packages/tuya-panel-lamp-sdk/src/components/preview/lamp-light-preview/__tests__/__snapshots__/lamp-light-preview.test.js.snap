// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LampLightPreview components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 640,
        "justifyContent": "flex-start",
        "width": 640,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 640,
          "width": 640,
        }
      }
    />
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 286,
            "position": "absolute",
            "width": 420,
          },
          Object {
            "bottom": 0,
            "left": 110,
            "tintColor": "rgba(255, 162, 71, 0.9)",
          },
        ]
      }
    />
  </View>
</View>
`;
